homeassistant/components/plaato/config_flow.py

```diff
@@ -1,10 +1,225 @@
-"""Config flow for GPSLogger."""
-from homeassistant.helpers import config_entry_flow
+"""Config flow for Plaato."""
+import logging
 
-from .const import DOMAIN
+from pyplaato.plaato import PlaatoDeviceType
+import voluptuous as vol
 
-config_entry_flow.register_webhook_flow(
-    DOMAIN,
-    "Webhook",
-    {"docs_url": "https://www.home-assistant.io/integrations/plaato/"},
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_SCAN_INTERVAL, CONF_TOKEN, CONF_WEBHOOK_ID
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_CLOUDHOOK,
+    CONF_DEVICE_NAME,
+    CONF_DEVICE_TYPE,
+    CONF_USE_WEBHOOK,
+    DEFAULT_SCAN_INTERVAL,
+    DOCS_URL,
+    PLACEHOLDER_DEVICE_NAME,
+    PLACEHOLDER_DEVICE_TYPE,
+    PLACEHOLDER_DOCS_URL,
+    PLACEHOLDER_WEBHOOK_URL,
 )
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__package__)
+
+
+class PlaatoConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handles a Plaato config flow."""
+
+    VERSION = 2
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize."""
+        self._init_info = {}
+        self._errors = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle user step."""
+
+        if user_input is not None:
+            self._init_info[CONF_DEVICE_TYPE] = PlaatoDeviceType(
+                user_input[CONF_DEVICE_TYPE]
+            )
+            self._init_info[CONF_DEVICE_NAME] = user_input[CONF_DEVICE_NAME]
+
+            return await self.async_step_api_method()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_DEVICE_NAME,
+                        default=self._init_info.get(CONF_DEVICE_NAME, None),
+                    ): str,
+                    vol.Required(
+                        CONF_DEVICE_TYPE,
+                        default=self._init_info.get(CONF_DEVICE_TYPE, None),
+                    ): vol.In(list(PlaatoDeviceType)),
+                }
+            ),
+            errors=self._errors,
+        )
+
+    async def async_step_api_method(self, user_input=None):
+        """Handle device type step."""
+
+        device_type = self._init_info[CONF_DEVICE_TYPE]
+
+        if user_input is None:
+            return await self._show_api_method_form(device_type)
+
+        token = user_input.get(CONF_TOKEN, None)
+        use_webhook = user_input.get(CONF_USE_WEBHOOK, False)
+
+        if not token and not use_webhook:
+            self._errors["base"] = PlaatoConfigFlow._get_error(device_type)
+            return await self._show_api_method_form(device_type)
+
+        self._init_info[CONF_USE_WEBHOOK] = use_webhook
```

Comment:
We should make sure this is always set to true or false

Reply:
It should always be se to false if not set at all. So it is always `False` or `True` since it is not possible to skip this step in the config flow.

```
use_webhook = user_input.get(CONF_USE_WEBHOOK, False)
```